x-pack/plugins/lists/server/services/exception_lists/utils.ts

```diff
@@ -316,61 +304,74 @@ export const transformUpdateCommentsToComments = ({
   existingComments: CommentsArray;
   user: string;
 }): CommentsArray => {
-  const newComments = comments ?? [];
-
-  if (newComments.length < existingComments.length) {
+  if (comments == null && existingComments.length > 0) {
+    // cannot delete, only append at the moment
+    // TODO: Add new comments endpoint to fully support CRUD
     throw new ErrorWithStatusCode(
-      'Comments cannot be deleted, only new comments may be added',
-      403
+      'Empty "[]" passed to "comments" - detected existing comments. Comment cannot be deleted.',
+      401
     );
-  } else {
-    return newComments.flatMap((c, index) => {
-      const existingComment = existingComments[index];
+  }
 
-      if (commentsSchema.is(existingComment) && !commentsSchema.is(c)) {
-        throw new ErrorWithStatusCode(
-          'When trying to update a comment, "created_at" and "created_by" must be present',
-          403
-        );
-      } else if (existingComment == null && commentsSchema.is(c)) {
-        throw new ErrorWithStatusCode('Only new comments may be added', 403);
-      } else if (
-        commentsSchema.is(c) &&
-        existingComment != null &&
-        isCommentEqual(c, existingComment)
-      ) {
-        return existingComment;
-      } else if (commentsSchema.is(c) && existingComment != null) {
-        return transformUpdateComments({ comment: c, existingComment, user });
-      } else {
-        return transformCreateCommentsToComments({ comments: [c], user }) ?? [];
-      }
-    });
+  if (comments != null && comments.length < existingComments.length) {
+    // cannot delete, only append at the moment
+    // TODO: Add new comments endpoint to fully support CRUD
+    throw new ErrorWithStatusCode('Comment cannot be deleted.', 401);
   }
+
+  if (comments == null && existingComments.length === 0) {
+    return [];
+  }
+
+  const incomingComments = comments ?? [];
+
+  return incomingComments.map<Comment>((item, index) => {
+    const matchingCommentByIndex = existingComments[index];
+    const idMatches = matchingCommentByIndex && item.id === matchingCommentByIndex.id;
+
+    // cannot delete, only append at the moment
+    // TODO: Add new comments endpoint to fully support CRUD
+    if (!idMatches && index < existingComments.length) {
+      throw new ErrorWithStatusCode(`Cannot delete comment id: ${matchingCommentByIndex.id}`, 401);
```

Comment:
I think we're overusing the `401` error code in this file. This seems to be causing unexpected side effects like the user getting logged out of Kibana. We should consider using a different error code like `400` for these types of errors. @FrankHassanabad thoughts?

Reply:
Yes, these probably should be 400s. 401 should only be used on failure to authenticate, 403 for unauthorized accesses.

Reply:
Thanks @peluja1012 and @madirey ! I simplified validations and made comments append only. Full CRUD is a wip for now.